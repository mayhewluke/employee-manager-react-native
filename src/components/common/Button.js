import React from "react";
import { TouchableOpacity, Text } from "react-native";
import PropTypes from "prop-types";

const styles = {
  buttonStyle: {
    flex: 1,
    alignSelf: "stretch",
    backgroundColor: "#fff",
    borderRadius: 5,
    borderWidth: 1,
    borderColor: "#007aff",
    marginLeft: 5,
    marginRight: 5
  },
  textStyle: {
    alignSelf: "center",
    color: "#007aff",
    fontSize: 16,
    fontWeight: "600",
    paddingTop: 10,
    paddingBottom: 10
  }
};

const Button = ({ onPress, children }) => {
  const { buttonStyle, textStyle } = styles;
  return (
    <TouchableOpacity style={buttonStyle} onPress={onPress}>
      <Text style={textStyle}>{children}</Text>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  onPress: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export { Button };
