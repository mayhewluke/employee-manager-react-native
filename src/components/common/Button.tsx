import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

const styles = StyleSheet.create({
  buttonStyle: {
    backgroundColor: "#fff",
    borderColor: "#007aff",
    borderRadius: 5,
    borderWidth: 1,
    marginLeft: 5,
    marginRight: 5,
  },
  textStyle: {
    alignSelf: "center",
    color: "#007aff",
    fontSize: 16,
    fontWeight: "600",
    paddingBottom: 10,
    paddingTop: 10,
  },
});

const Button: React.SFC<{ onPress: () => void }> = ({ onPress, children }) => {
  const { buttonStyle, textStyle } = styles;
  return (
    <TouchableOpacity style={buttonStyle} onPress={onPress}>
      <Text style={textStyle}>{children}</Text>
    </TouchableOpacity>
  );
};

export default Button;
