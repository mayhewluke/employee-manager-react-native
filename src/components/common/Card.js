import React from "react";
import { View } from "react-native";
import PropTypes from "prop-types";

const styles = {
  containerStyle: {
    borderWidth: 1,
    borderRadius: 2,
    borderColor: "#ddd",
    borderBottomWidth: 0,
    elevation: 1,
    marginLeft: 5,
    marginRight: 5,
    marginTop: 10
  }
};

const Card = ({ children }) => {
  const { containerStyle } = styles;
  return <View style={containerStyle}>{children}</View>;
};

Card.propTypes = {
  children: PropTypes.node.isRequired
};

export default Card;
