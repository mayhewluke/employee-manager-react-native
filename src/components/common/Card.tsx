import React from "react";
import { StyleSheet, View } from "react-native";

const styles = StyleSheet.create({
  containerStyle: {
    borderBottomWidth: 0,
    borderColor: "#ddd",
    borderRadius: 2,
    borderWidth: 1,
    elevation: 1,
    marginLeft: 5,
    marginRight: 5,
    marginTop: 10,
  },
});

const Card: React.SFC<{}> = ({ children }) => {
  const { containerStyle } = styles;
  return <View style={containerStyle}>{children}</View>;
};

export default Card;
