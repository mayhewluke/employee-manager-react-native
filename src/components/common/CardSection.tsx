import React from "react";
import { StyleSheet, View, ViewProps } from "react-native";

const styles = StyleSheet.create({
  containerStyle: {
    backgroundColor: "#fff",
    borderBottomWidth: 1,
    borderColor: "#ddd",
    flexDirection: "row",
    justifyContent: "flex-start",
    padding: 5,
    position: "relative",
  },
});

const CardSection: React.SFC<ViewProps> = ({ children, style, ...rest }) => (
  <View style={[styles.containerStyle, style]} {...rest}>
    {children}
  </View>
);

export default CardSection;
