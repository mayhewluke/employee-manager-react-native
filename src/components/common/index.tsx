export { default as AsyncButton } from "./AsyncButton";
export { default as Button } from "./Button";
export { default as Card } from "./Card";
export { default as CardSection } from "./CardSection";
export { default as Confirm } from "./Confirm";
export { default as Input } from "./Input";
export { default as Spinner, Size as SpinnerSize } from "./Spinner";
