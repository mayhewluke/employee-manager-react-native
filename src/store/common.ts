export type Async<T> =
  | { state: "INIT" }
  | { state: "PROGRESS" }
  | { state: "COMPLETE"; value: T }
  | { state: "ERROR"; error: string };
