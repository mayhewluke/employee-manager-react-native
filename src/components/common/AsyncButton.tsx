import * as React from "react";
import { StyleSheet, Text, View } from "react-native";

import { Async } from "../../store";
import Button from "./Button";
import Spinner, { Size as SpinnerSize } from "./Spinner";

interface IProps {
  label: string;
  asyncAction: Async<any>;
  onPress: () => void;
  size?: SpinnerSize;
}

const styles = StyleSheet.create({
  errorTextStyle: {
    alignSelf: "center",
    color: "red",
    fontSize: 20,
  },
});

const AsyncButton: React.SFC<IProps> = ({
  label,
  asyncAction,
  onPress,
  size = SpinnerSize.Large,
}) => (
  <View style={{ flex: 1, flexDirection: "row" }}>
    {asyncAction.state === "ERROR" && (
      <Text style={styles.errorTextStyle}>{asyncAction.error}</Text>
    )}
    {asyncAction.state === "PROGRESS" ? (
      <Spinner size={size} />
    ) : (
      <Button onPress={onPress}>{label}</Button>
    )}
  </View>
);

export default AsyncButton;
