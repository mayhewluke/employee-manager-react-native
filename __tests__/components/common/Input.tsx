import "jest-enzyme";
import React from "react";

import { shallow } from "enzyme";

import { Input } from "../../../src/components/common";

// Some components show up in snapshots as <Component> if we don't mock them
jest.mock("react-native", () => ({
  StyleSheet: { create: jest.fn(obj => obj) },
  Text: "Text",
  TextInput: "TextInput",
  View: "View",
}));

describe("Input", () => {
  const defProps = {
    label: "Enter text here",
  };
  const mkWrapper = (propOverrides?: any) => {
    const props = { ...defProps, ...propOverrides };
    return shallow(<Input {...props} />);
  };

  it("renders correctly", () => {
    expect(mkWrapper()).toMatchSnapshot();
  });

  it("passes all props down to the TextInput", () => {
    const props = { foo: "bar" };
    expect(
      mkWrapper(props)
        .find("TextInput")
        .prop("foo"),
    ).toEqual(props.foo);
  });
});
