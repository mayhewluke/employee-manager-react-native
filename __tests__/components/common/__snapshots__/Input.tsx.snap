// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 40,
    }
  }
>
  <Text
    style={
      Object {
        "flex": 1,
        "fontSize": 18,
        "paddingLeft": 20,
      }
    }
  >
    Enter text here
  </Text>
  <TextInput
    label="Enter text here"
    style={
      Object {
        "borderColor": "#000",
        "borderWidth": 1,
        "color": "#000",
        "flex": 2,
        "fontSize": 18,
        "lineHeight": 23,
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  />
</View>
`;
